Objects/call.c

```diff
@@ -766,30 +766,36 @@ PyEval_CallObjectWithKeywords(PyObject *callable,
     assert(!PyErr_Occurred());
 #endif
 
-    if (args == NULL) {
-        return _PyObject_FastCallDict(callable, NULL, 0, kwargs);
-    }
-
-    if (!PyTuple_Check(args)) {
+    if (args != NULL && !PyTuple_Check(args)) {
```

Comment:
Please update the commit message which doesn't make sense compared to the modified code.

This commit fixes a bug, no? It makes sure that a TypeError is raised if kwargs is not a dictionary, right? Please explain it in the commit message.